python_transport/wirepas_gateway/configure_node.py

```diff
@@ -0,0 +1,339 @@
+# Copyright 2020 Wirepas Ltd licensed under Apache License, Version 2.0
+#
+# See file LICENSE for full license details.
+#
+import binascii
+import argparse
+
+from enum import Enum
+
+from wirepas_gateway.dbus.dbus_client import BusClient
+
+
+class NodeRole:
+    """Utility class to package a node role
+
+    Helper to manipulate and convert node role based on a Base role
+    and a list of flags
+
+    """
+
+    class BaseRole(Enum):
+        SINK = 0x1
+        ROUTEUR = 0x2
+        NON_ROUTEUR = 0x3
```

Comment:
Not sure it is usefull but do you want to add advertiser role ?

Reply:
Yes:

```
$ wm-node-conf set -s sink0 -r "sink autorole"
Cannot set 129 for param node_role on sink sink0: GDBus.Error:com.wirepas.sink.config.error: [WPC_set_role]: C Mesh Lib ret = 4
Configuration done with result = GatewayResultCode.GW_RES_INVALID_PARAM